docs/docs/demos/secops.md

```diff
@@ -87,3 +87,80 @@ erDiagram
     artifact ||--|| pipeline : "built by pipeline"
 ```
+
+## Program
+
+The E-R diagram above is the schema for the data as provided.  We want
+to know not just the vulnerabilities in source trees from specific
+pipelines, as the `vulnerability` table informs us, but also in
+everything derived from those: in `pipelin_source`s, `artifact`s
+(directly and indirectly), `k8sobject`s, and `k8scluster`s.  Finally,
+we want to track vulnerability statistics.
+
+If we implement all of that as SQL views, we end up with the following
+structure, which shows tables as before with views added as names in
+upper case:
+
+```mermaid
+erDiagram
+    pipeline ||--|| pipeline_sources : "has sources"
+    pipeline_sources ||--o{ git_commit : "from Git commit"
+    git_commit ||--|| repository : "within Git repository"
+    k8sobject ||--|| artifact : "built from"
+    k8sobject ||--o{ k8scluster : "deployed as"
+    vulnerability ||--|| pipeline_sources : "discovered in"
+    artifact ||--o| artifact : "parent artifact"
+    artifact ||--|| pipeline : "built by pipeline"
+    PIPELINE_VULNERABILITY ||--|| pipeline_sources : ""
+    PIPELINE_VULNERABILITY ||--|| vulnerability : ""
+    ARTIFACT_VULNERABILITY ||--|| artifact : ""
+    ARTIFACT_VULNERABILITY ||--|| PIPELINE_VULNERABILITY : ""
+    TRANSITIVE_ARTIFACT_VULNERABILITY ||--|| artifact : ""
+    TRANSITIVE_ARTIFACT_VULNERABILITY ||--|| ARTIFACT_VULNERABILITY : ""
+    K8SOBJECT_VULNERABILITY ||--|| k8sobject : ""
+    K8SOBJECT_VULNERABILITY ||--|| TRANSITIVE_ARTIFACT_VULNERABILITY : ""
+    K8SCLUSTER_VULNERABILITY ||--|| k8sobject : ""
+    K8SCLUSTER_VULNERABILITY ||--|| K8SOBJECT_VULNERABILITY : ""
+    K8SOBJECT_VULNERABILITY_STATS ||--|| K8SCLUSTER_VULNERABILITY : ""
+    K8SOBJECT_VULNERABILITY_STATS ||--|| vulnerability : ""
+    K8SOBJECT_VULNERABILITY_STATS ||--|| k8scluster : ""
+
+```
+
+## Program and Pipeline
+
+The demo includes a program and a pipeline.  To see the SQL for these
+tables and views, start the DBSP demo, visit <http://localhost:8085/>,
+click on Existing Programs in the sidebar, and then click on the
+pencil icon next to SecOps demo.  It shows a SQL program with a
+`CREATE TABLE` command for each table and a `CREATE VIEW` command for
+each view.
+
+The demo also includes a pipeline that uses this program.  On the
+sidebar, click on Pipeline Management, then on the pencil icon next to
+SecOps Pipeline.  This displays the pipeline structure, including the
+program and how it is attached to connectors:
+
+![SecOps Pipeline](images/secops-pipeline.png)
+
+This demo has significantly more tables and views than the other
+examples.  This demo attaches some of them to connectors and it leaves
+others unattached.  The unattached tables would provide data that is
+not needed for the particular analysis that we are doing, and the
+unattached views are ones that we are not yet interested in monitoring
+on output.  In fact, only a single view is attached to a connector,
```

Comment:
have we ever checked that if we attach them they work?

Reply:
I don't know a way to do that. The UI doesn't work for views in general right now. My understanding is that @gz put something in so that this one specific view in the secops demo does work. I have written the docs on the speculation that views will work and that then the docs about how to use them will be useful.

Reply:
@gz is this still broken?
What should we do?

Reply:
>  What should we do?

I think we should just be patient and let @gz work on it.